Source/Core/DolphinQt/GCMemcardManager.cpp

```diff
@@ -447,17 +447,35 @@ void GCMemcardManager::DrawIcons()
   const auto column = 3;
   for (int slot = 0; slot < SLOT_COUNT; slot++)
   {
-    int row = 0;
-    for (auto& icon : m_slot_active_icons[slot])
+    // skip loop if the table is empty
+    if (m_slot_table[slot]->rowCount() <= 0)
+      continue;
+
+    const auto viewport = m_slot_table[slot]->viewport();
+    auto row = m_slot_table[slot]->indexAt(viewport->rect().topLeft()).row();
+    const auto max_table_index = m_slot_table[slot]->indexAt(viewport->rect().bottomLeft());
+    const auto max_row =
+        max_table_index.row() < 0 ? (m_slot_table[slot]->rowCount() - 1) : max_table_index.row();
+
+    for (; row <= max_row; row++)
     {
-      int frame = (m_current_frame / 3 - icon.first) % icon.second.size();
+      auto& icon = m_slot_active_icons[slot][row];
+
+      // this icon doesn't have an animation
+      if (icon.second.size() <= 1)
+        continue;
+
+      const auto prev_frame = ((m_current_frame - 1) / 3 - icon.first) % icon.second.size();
+      const auto frame = (m_current_frame / 3 - icon.first) % icon.second.size();
 
+      if (prev_frame == frame)
+        continue;
+      
```

Comment:
Lint is complaining about this line because it has a space/tab.

Reply:
ye i noticed. its a space. made me scratch my head for a second there as it was suggesting a delete and insert of a new line xD

Reply:
ye i noticed, its a manual tab. not sure how it got there but it made me scratch my head for a second there
'delete line and insert new line' xD